File: 프로그래머스/2/42584. 주식가격/주식가격.py
# def solution(prices):
#     #0650
#     #prices 돌면서, 과거가격과 지금가격 비교해서 얼마나 
#     answer = []
#     priceLength=len(prices)
#     for i in range (priceLength):
#         count=0
#         if i== priceLength-1: 
#             answer.append(count) 
#             break #마지막이면 count 추가하고 for i 반복문 끝냄
#         for j in range (i+1, priceLength):
#             count+=1
#             if prices[i] >prices[j] or j== priceLength-1: #상승중단 혹은 j가 끝까지 가면 count추가후 for j반복문 break
#                 answer.append(count)
#                 # print(i,j)
#                 break
#             # if i== priceLength-1:break
#             # if j== priceLength-1:
#             #     answer.append(count)
#     return answer

def solution(prices):
    answer = []
    for i in range(len(prices) - 1):
        sec = 0
        for j in range(i, len(prices) - 1):
            if(prices[i] <= prices[j]):
                sec += 1
            else:
                break
        answer.append(sec);
    answer.append(0);
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
// function solution(prices) {
//     var answer = [];
//     for (let i=0;i<prices.length;i++){
//         let count=1
//         for(let j=i+1 ;j<prices.length;j++){
//             if (j!=prices.length-1 && prices[i]<=prices[j]){
//                 count++
//             }else{
//                 answer.push(count)
//                 break
//             }
//         }
//     }
//     answer.push(0)
//     return answer;
// }



// function solution(prices) {
//     const answer = [];
//     for (let i = 0; i < prices.length; i++) {
//         let stack = 0;
//         for (let j = i + 1; j < prices.length; j++) {
//             stack++;
//             if (prices[i] > prices[j]) {
//                 break;
//             }
//         }
//         answer.push(stack);
//     }
//     return answer;
// }

//https://kangwlgns.tistory.com/30


function solution(prices){
    let answer=Array(prices.length).fill(0)
    prices.forEach((el,idx)=>{
        if(idx===prices.length-1) {
            answer[idx]=0
            return
        }
        let count =1
        for(let i= idx+1; i<prices.length; i++){
            if(prices[i] < el) break
            count++
            if(i === prices.length-1) count-=1
        }
        // answer.push(count)
        answer[idx]=count
    })
    
    return answer
}